### ISSUE TITLE
[FEATURE] Implement Alerts and Notifications from Rancher for Longhorn

### ISSUE THREAD

**Issue by Sheldonwl** [longhorn/longhorn#1473]
**Is your feature request related to a problem? Please describe.**
This feature request relates to: https://github.com/longhorn/longhorn/issues/928https://github.com/longhorn/longhorn/issues/928
It would be nice to setup Alerts and Notifications in Rancher to get notified about certain things in Longhorn. 

**Describe the solution you'd like**
To be able to create Alerts from within Rancher and attached Notifiers to those Alerts. This would require an integration with Rancher. 

**Describe alternatives you've considered**
Being able to expose and export logs from Longhorn to a central logging tool. This would also be able to trigger alerts, if that tool support such a feature.

**Comment by yasker** [comment]
@Sheldonwl  I think the user can already do this in v1.1.0, as a part of https://github.com/longhorn/longhorn/issues/1180 , e.g. https://longhorn.io/docs/1.1.0/monitoring/integrating-with-rancher-monitoring/ Can you confirm it's done?

**Comment by amnk** [comment]
@yasker what do you think about having ServiceMonitor object support in Longhorn's chart? I can create a PR to enable that if you think this is a good idea.

**Comment by yasker** [comment]
@amnk Unfortunately I am not super familiar with this part. @PhanLe1010 what do you think?

**Comment by PhanLe1010** [comment]
@amnk 

Do you mean deploying this service monitoring CR with Longhorn by default? https://longhorn.io/docs/1.1.2/monitoring/integrating-with-rancher-monitoring/#add-longhorn-metrics-to-the-rancher-monitoring-system

It would mean that Longhorn also must be shipped with `ServiceMonitor` CRD depending on where user cluster already has it or not. IMO, it is best to just let user deploy the service monitor CR by themself if users want to enable monitoring. Similar to what we did with CSI snapshot.

**Comment by amnk** [comment]
@PhanLe1010 lately a lot of charts include `Servicemonitor` object in the chart. For example [nginx-ingress](https://github.com/kubernetes/ingress-nginx/blob/main/charts/ingress-nginx/templates/controller-servicemonitor.yaml) does that, as well as [metallb](https://github.com/bitnami/charts/blob/master/bitnami/metallb/templates/controller/servicemonitor.yaml), [coredns](https://github.com/coredns/helm/blob/master/stable/coredns/templates/servicemonitor.yaml), [VictoriaMetrics](https://github.com/VictoriaMetrics/helm-charts/blob/master/charts/victoria-metrics-cluster/templates/vminsert-service-monitor.yaml) and a lot more. 

Of course user can deploy that himself, but including that in the default chart just makes user experience better (imho)

**Comment by ybykov-a9s** [comment]
@PhanLe1010 any changes here? It is a great idea to configure a ServiceMonitor in the Helm chart as a lot of other Helm charts do. Of course it can be disabled by default, But it allows users by using only one option enable it. As well as enabling ability to define Prometheus rules is also a great possibility to simplify overall configuration process.

**Comment by PhanLe1010** [comment]
Thanks @amnk and @ybykov-a9s  for the feedback.

We will discuss this proposal with the team 

cc @joshimoo @shuo-wu @innobead. Do you have any opinion on this? I agree with users that we should include the service monitor and example Prometheus rules in the Helm chart but disable it by default.

**Comment by jenting** [comment]
> Thanks @amnk and @ybykov-a9s  for the feedback. We will discuss this proposal with the team 
> 
> 
> 
> cc @joshimoo @shuo-wu @innobead. Do you have any opinion on this? I agree with users that we should include the service monitor and example Prometheus rules in the Helm chart but disable it by default. 

+1

**Comment by joshimoo** [comment]
@PhanLe1010 I agree if we can make the deployment toggle + user configurable during helm / rancher chart installation and it does not impact upgrade process, it would be a nice to have to expose + ship it. 

We can then use the same approach to ship additional optional stuff during installation like network policy, pprof scraping/monitoring, etc down the line.

cc @innobead what do you think?

**Comment by innobead** [comment]
100% agree.

**Comment by w13915984028** [comment]
connect with https://github.com/harvester/harvester/issues/2517
Harvester is planning to add logging and alert for all related componenets.

**Comment by MSandro** [comment]
any news here?
